src/CppUTest/Utest.cpp

```diff
@@ -486,9 +488,13 @@ void UtestShell::print(const SimpleString& text, const char* fileName, int lineN
     print(text.asCharString(), fileName, lineNumber);
 }
 
+extern "C++" {
+
```

Comment:
There would be lots more of strange things like this to make it compile with Gcc and Clang.